fig_test.go

```diff
@@ -1263,3 +1280,30 @@ func setenv(t *testing.T, key, value string) {
 	t.Helper()
 	t.Setenv(key, value)
 }
+
+func (l *ListenerType) UnmarshalString(v string) error {
+	switch strings.ToLower(v) {
+	case "unix":
+		*l = ListenerUnix
+	case "tcp":
+		*l = ListenerTCP
+	case "tls":
+		*l = ListenerTLS
+	default:
+		return fmt.Errorf("unknown listener type: %s", v)
+	}
+	return nil
+}
+
+func (l ListenerType) String() string {
+	switch l {
+	case ListenerUnix:
+		return "unix"
+	case ListenerTCP:
+		return "tcp"
+	case ListenerTLS:
+		return "tls"
+	default:
+		return "unknown"
+	}
+}
```

Comment:
Is this used somewhere?

Reply:
It was used in an error message output, but i've removed it and switched from `%s` to `%d`